### PR TITLE
297 translator errors

### PR DESCRIPTION
**Pull request by charles-turner-1**
- Added error handling 
- Draft for a decorator that should keep the error handling all in one place - having some issues getting the correct function name to pass through though.

### CODE REVIEW

File: src/access_nri_intake/catalog/utils.py [hunk 1]
```diff
@@ -26,3 +27,28 @@ def wrapper(*args, **kwargs):
...
 
     return wrapper
+
+
+def trace_failure(func: Callable) -> Callable:
+    """
+    Decorator that wraps a function and prints a message if it raises an exception
+    """
+    func_name = func.__name__
+    colname = func_name[1:].split("_")[0]
```
Comment (marc-white) [inline src/access_nri_intake/catalog/utils.py:37]: Exactly what issue are you having with getting the function name out?

Comment (charles-turner-1) [reply src/access_nri_intake/catalog/utils.py:37]: `func.__name__` will return `wrapper` here - seems to be a common problem eg. https://stackoverflow.com/questions/50350379/getting-function-name-from-nested-decorator, but none of the solutions I've found seem to have solved it yet.

Comment (charles-turner-1) [reply src/access_nri_intake/catalog/utils.py:37]: Fixed by changing the order of decoration & moving definitions in - about to tidy up & push.

Comment (marc-white) [reply src/access_nri_intake/catalog/utils.py]: Cool, if you take it out of Draft I can approve it so it can go in

**Comment by charles-turner-1** [comment]
Codecov is complaining that this decreases project coverage - but the coverage on the diff is 100% so...

**Comment by marc-white** [comment]
> Codecov is complaining that this decreases project coverage - but the coverage on the diff is 100% so...

Yeah, I think we just ignore that.

**Review by marc-white — APPROVED** [review]
Looks good, approved to merge. I'll override the branch protection and merge it.

**Pull request merged**